packages/kit/src/runtime/server/index.js

```diff
@@ -152,7 +152,9 @@ export async function respond(incoming, options, state = {}) {
 									};
 								}
 
-								response.headers['etag'] = etag;
+								if (!response.headers['etag']) {
+									response.headers['etag'] = etag;
+								}
```

Comment:
Shouldn't we be bailing out on line 121? If there's already an ETag then surely we should never get to this point?

```suggestion
								response.headers['etag'] = etag;
```

Feels like we need to make this change instead:

```diff
-if (response.status === 200) {
+if (response.status === 200 && !response.headers['etag']) {

Reply:
made this change